### ISSUE TITLE
`getParentRevision` returns invalid revision when parent is a merge commit

### ISSUE THREAD

**Issue by OliverJAsh** [paularmstrong/build-tracker#137]
All of our commits on our `master` branch are merge commits, which means they have two parents. Because of this, `getParentRevision` returns two commits when it's expected to only return one:

```sh-session
$ git log --pretty=%P -n 1 master
143abcfc1c4416d2f1d045f309aadbd7baf2f7b2 ece9964638c730e9be046c6d23187b9b57b243bd
```

This in turn means we get an error when trying to upload the build, because the parent revision is too long to fit into the database (max characters is 64).

What should happen in this case? I guess we have to pick one of the merge commit's parents to act as the "main parent".

**Comment by paularmstrong** [comment]
Interesting. I hadn't thought about this use case—I'm used to working in environments that require squash & rebase on master so that we don't end up with confusing merge commits.

If I'm reading the docs correctly, the first parent given is the one that we'd want to use for comparison. Can you give more details of those SHAs: was each of them built and uploaded to BT? Was either a "merge commit"? Do both have changes in them?

**Comment by OliverJAsh** [comment]
When merging a PR into `master`, the new merge commit's first parent will be the previous commit on `master` (which could also be a merge commit) and the second parent will be the `HEAD` of the PR that's was merged—IIUC. So yeah, I believe both of those SHAs would be uploaded to BT.

I was able to workaround this issue by using the first parent, e.g. `split(' ')[0]`. Perhaps we should just do that?
